plugins/notifiers/client.go

```diff
@@ -29,16 +42,89 @@ func (c SlackConfig) Decode(v interface{}) error {
 	return mapstructure.Decode(c, v)
 }
 
+// LarkConfig is a map of workspace name to config
+type MultiConfig map[string]interface{}
+
+func (c MultiConfig) Decode(v interface{}) error {
+	return mapstructure.Decode(c, v)
+}
+
+type ClientConfig map[string]interface {
+	// AccessToken  string `mapstructure:"access_token"`
+	// ClientId     string `mapstructure:"client_id"`
+	// ClientSecret string `mapstructure:"client_secret"`
+}
+
+// LarkConfig is a map of workspace name to config
+type SlackLarkConfig map[string]interface{}
+
+func (c SlackLarkConfig) Decode(v interface{}) error {
+	return mapstructure.Decode(c, v)
+}
+
 type Config struct {
-	Provider string `mapstructure:"provider" validate:"omitempty,oneof=slack"`
+	Provider string `mapstructure:"provider" validate:"omitempty,oneof=slack lark"`
 
 	// slack
 	AccessToken string      `mapstructure:"access_token" validate:"required_without=SlackConfig"`
 	SlackConfig SlackConfig `mapstructure:"slack_config" validate:"required_without=AccessToken,dive"`
+	// custom messages
+	Messages domain.NotificationMessages
+}
 
+type ConfigMultiClient struct {
+	Notifiers map[string]Notifier `mapstructure:"notifiers"`
 	// custom messages
 	Messages domain.NotificationMessages
 }
+type Notifier struct {
+	Provider     string `mapstructure:"provider"`
+	AccessToken  string `mapstructure:"access_token,omitempty"`
+	ClientID     string `mapstructure:"client_id,omitempty"`
+	ClientSecret string `mapstructure:"client_id,omitempty"`
+	Criteria     string `mapstructure:"criteria"`
+}
+
+func NewMultiClient(config *ConfigMultiClient, logger log.Logger) (NotifyManager, error) {
+	fmt.Println("new client multi")
+	fmt.Println(config.Notifiers)
+	notifyManager := &NotifyManager{}
+	for key, notifier := range config.Notifiers {
+		fmt.Printf("Key: %s, Value: %+v\n", key, notifier)
+		if notifier.Provider == ProviderTypeSlack {
+
+			httpClient := &http.Client{Timeout: 10 * time.Second}
+
+			slackConfig, err := GetSlackConfig(&notifier, config.Messages)
+			if err != nil {
+				fmt.Println("lark send config 2" + notifier.Provider + err.Error())
```

Comment:
if there's a notifier misconfiguration, we should return the error and fail the server initialization altogether
```suggestion
				return nil, fmt.Errorf("failed to initialize %q notifier: %w", key, err)
```